equinix/resource_fabric_service_profile_acc_test.go

```diff
@@ -17,139 +17,138 @@ import (
 	v4 "github.com/equinix-labs/fabric-go/fabric/v4"
 )
 
-func testAccFabricReadServiceProfileConfig(uuid string) string {
-	return fmt.Sprintf(`data "equinix_fabric_service_profile" "test" {
-	uuid = "%s"
-	}
-`, uuid)
-}
+func TestAccFabricCreateServiceProfile_PFCR(t *testing.T) {
+	ports := GetFabricEnvPorts(t)
 
-func testAccFabricReadServiceProfilesListConfig(name string) string {
-	return fmt.Sprintf(`data "equinix_fabric_service_profiles" "test" {
-		filter {
-			property = "/name"
-			operator = "="
-			values = ["%s"]
-		}
+	var portUuidDot1Q, portMetroCodeDot1Q string
+	var portUuidQinq, portMetroCodeQinq string
+	if len(ports) > 0 {
+		portUuidDot1Q = ports["pfcr"]["dot1q"][0].Uuid
+		portMetroCodeDot1Q = ports["pfcr"]["dot1q"][0].Location.MetroCode
+		portUuidQinq = ports["pfcr"]["qinq"][0].Uuid
+		portMetroCodeQinq = ports["pfcr"]["qinq"][0].Location.MetroCode
 	}
-`, name)
-}
 
-func TestAccFabricReadServiceProfileByUuid(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { acceptance.TestAccPreCheck(t) },
-		Providers: acceptance.TestAccProviders,
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: checkServiceProfileDelete,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccFabricReadServiceProfileConfig("bfb74121-7e2c-4f74-99b3-69cdafb03b41"),
+				Config: testAccFabricCreateServiceProfileConfig(portUuidDot1Q, "XF_PORT", portMetroCodeDot1Q),
```

Comment:
You should get your port type directly from the Port ENV variable. The point is so that we don't have to update the strings here if we have to update the Port ENV variable. Add another variable to the list above and use it here.

Reply:
done